File: uiux_preview/snap.py
import asyncio
import os
from playwright.async_api import async_playwright

OUT_DIR = "/workspace/uiux_preview/screenshots"
os.makedirs(OUT_DIR, exist_ok=True)

URL = "http://localhost:8123/"

# We will use a wide viewport (1440) and full-page screenshot
# Plus component-specific captures
SHOTS = [
    # (filename, viewport, selector, full_page, extra_action)
    ("00-full", (1440, 3000), None, True, None),
    ("01-masthead", (1440, 900), ".masthead", False, None),
    ("02-ch1-library-pair", (1440, 1300), "#ch1 .pair", False, None),
    ("03-ch1-doctile-component", (1440, 900), "#ch1 .component-block", False, None),
    ("04-ch2-reader-pair", (1440, 1500), "#ch2 .pair", False, None),
    ("05-ch2-toc-component", (1440, 900), "#ch2 .component-block", False, None),
    ("06-ch3-settings-pair", (1440, 1500), "#ch3 .pair", False, None),
    ("07-ch4-liquid-glass", (1440, 700), "#ch4 .component-canvas", False, None),
    ("08-ch5-typography", (1440, 800), "#ch5 .component-block", False, None),
    ("09-ch6-color-palette", (1440, 900), "#ch6 .component-block", False, None),
]

async def main():
    async with async_playwright() as p:
        browser = await p.chromium.launch()
        for filename, (w, h), selector, full_page, _ in SHOTS:
            ctx = await browser.new_context(viewport={"width": w, "height": h}, device_scale_factor=2)
            page = await ctx.new_page()
            await page.goto(URL, wait_until="networkidle")
            # wait for fonts
            await page.wait_for_timeout(800)
            if selector:
                el = await page.query_selector(selector)
                if not el:
                    print(f"  ! selector not found: {selector}")
                else:
                    await el.scroll_into_view_if_needed()
                    await page.wait_for_timeout(200)
                    await el.screenshot(path=f"{OUT_DIR}/{filename}.png")
                    print(f"  ✓ {filename}.png")
            else:
                await page.screenshot(path=f"{OUT_DIR}/{filename}.png", full_page=full_page)
                print(f"  ✓ {filename}.png (full page)")
            await ctx.close()
        await browser.close()

asyncio.run(main())
print("DONE")

File: uiux_preview/snap.js
const { chromium } = require('/root/.npm/_npx/e41f203b7505f1fb/node_modules/playwright');
const fs = require('fs');
const path = require('path');

const OUT_DIR = '/workspace/uiux_preview/screenshots';
fs.mkdirSync(OUT_DIR, { recursive: true });

const URL = 'http://localhost:8123/';

const SHOTS = [
  ['00-full',          1440, 3200, null,         true],
  ['01-masthead',      1440,  900, '.masthead',  false],
  ['02-ch1-pair',      1440, 1300, '#ch1 .pair', false],
  ['03-ch1-doctile',   1440,  900, '#ch1 .component-block', false],
  ['04-ch2-pair',      1440, 1500, '#ch2 .pair', false],
  ['05-ch2-toc',       1440,  900, '#ch2 .component-block', false],
  ['06-ch3-pair',      1440, 1500, '#ch3 .pair', false],
  ['07-ch4-glass',     1440,  700, '#ch4 .component-canvas', false],
  ['08-ch5-type',      1440,  800, '#ch5 .component-block', false],
  ['09-ch6-color',     1440,  900, '#ch6 .component-block', false],
  // mobile-size captures for phone-pair blocks
  ['m1-library',        900, 1300, '#ch1 .pair',  false],
  ['m2-reader',         900, 1500, '#ch2 .pair',  false],
  ['m3-settings',       900, 1500, '#ch3 .pair',  false],
];

(async () => {
  const browser = await chromium.launch();
  for (const [name, w, h, sel, full] of SHOTS) {
    const ctx = await browser.newContext({
      viewport: { width: w, height: h },
      deviceScaleFactor: 2,
    });
    const page = await ctx.newPage();
    await page.goto(URL, { waitUntil: 'networkidle' });
    await page.waitForTimeout(900);
    if (sel) {
      const el = await page.$(sel);
      if (!el) {
        console.log(`  ! selector not found: ${sel}`);
      } else {
        await el.scrollIntoViewIfNeeded();
        await page.waitForTimeout(200);
        await el.screenshot({ path: path.join(OUT_DIR, `${name}.png`) });
        console.log(`  ✓ ${name}.png`);
      }
    } else {
      await page.screenshot({ path: path.join(OUT_DIR, `${name}.png`), fullPage: full });
      console.log(`  ✓ ${name}.png (full)`);
    }
    await ctx.close();
  }
  await browser.close();
  console.log('DONE');
})();
